canvas/cli/common.py

```diff
@@ -1,35 +1,51 @@
+import json
 import sys
 
+DEFAULT_JSON = False
+DEFAULT_TABLE = False
+
 # keys: [(items key, title, pretty title), ...]
-def cli_list(items, keys, table = False, skip_headers = False,
-        collective_name = 'items', sort_key = 'id'):
+def cli_list(items, keys, table = DEFAULT_TABLE, skip_headers = False,
```

Comment:
After a few iterations, we have not really been able to simplify this function too much (sometimes it even gets more complex).

How about we just split the functionality into three functions (i.e., _cli_list_plain(), _cli_list_table(), and _cli_list_json()).
Then, this top-level function can handle to initial sort and route to the correct function.
Then, you may not even have to store the values in an intermediate list
(but you still can if it makes the most sense).